src/platforms/rp2040/src/lib/gpiodriver.c

```diff
@@ -133,7 +141,22 @@ static term nif_gpio_digital_write(Context *ctx, int argc, term argv[])
             return ERROR_ATOM;
         }
     }
-    gpio_put(gpio_num, level);
+
+    int gpio_num;
+    if (term_is_integer(gpio_pin)) {
+        gpio_num = gpio_pin;
+        gpio_put(gpio_num, level);
+#ifdef LIB_PICO_CYW43_ARCH
+    } else if (term_is_atom(gpio_pin)) {
+        gpio_num = interop_atom_term_select_int(wl_pin_table, gpio_pin, ctx->global);
+        if (UNLIKELY((gpio_num == -1) || (gpio_num > 1))) {
+            return ERROR_ATOM;
+        }
+        cyw43_arch_gpio_put(gpio_num, level);
+#endif
+    } else {
+        return ERROR_ATOM;
```

Comment:
We should go through the semantic of the GPIO driver.  @UncleGrumpy and I noticed that the return type is `error` in a lot of cases, which doesn’t leave much for error reporting.

See Issue #894 

In this case, do you think we should raise a `badarg`?